src/nvim/ex_getln.c

```diff
@@ -1130,7 +1130,7 @@ static int command_line_handle_key(CommandLineState *s)
     if (!mouse_has(MOUSE_COMMAND)) {
       return command_line_not_changed(s);                   // Ignore mouse
     }
-    cmdline_paste(0, true, true);
+    cmdline_paste('*', true, true);
```

Comment:
comparing with `do_mouse` the register should be `eval_has_provider("clipboard") ? '*' : 0` (or an `if` if that looks better)